kythe/go/extractors/config/wrapper/gradle_build_modifier.go

```diff
@@ -0,0 +1,79 @@
+package wrapper
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+)
+
+// These are the lines necessary for gradle build to use a different javac.
+var kytheJavacWrapper = `
+allprojects {
+  gradle.projectsEvaluated {
+    tasks.withType(JavaCompile) {
+      options.fork = true
+      options.forkOptions.executable = '/opt/kythe/extractors/javac-wrapper.sh'
+    }
+  }
+}
+`
+
+// This matches a line which sets the javac to use Kythe's javac-wrapper.sh
+var kytheMatcher = regexp.MustCompile(`^\s*options\.forkOptions\.executable\ =\ '/opt/kythe/extractors/javac-wrapper.sh'\s*$`)
+
+// This matches any line which sets a new javac executable, useful for detecting
+// edge cases which already modify javac.
+var javacMatcher = regexp.MustCompile(`^\s*options\.forkOptions\.executable\ =.*$`)
+
+// PreProcessGradleBuild takes a gradle.build file and either verifies that it
+// already has the bits necessary to run kythe's javac wrapper, or adds that
+// functionality.
+//
+// Note this potentially modifies the input file, so make a copy beforehand if
+// you need to keep the original.
+func PreProcessGradleBuild(gradleBuildFile string) error {
+	k, err := hasKytheWrapper(gradleBuildFile)
+	if err != nil {
+		return err
+	}
+	if k {
+		// Already has the kythe javac-wrapper.
+		return nil
+	}
+	return appendKytheWrapper(gradleBuildFile)
+}
+
+func hasKytheWrapper(gradleBuildFile string) (bool, error) {
+	f, err := os.Open(gradleBuildFile)
```

Comment:
I think this'd be simpler if you slurp up the whole file:

```
bits, err := ioutil.ReadFile(path)
if err != nil {
  return false, err
}
if kytheMatcher.Match(bits) {
  return true, nil
}
if javacMatcher.Match(bits) {
  return false, fmt.Errorf(…)
}
return false, nil
```

Granted you'll get more than you need, but these files shouldn't be so big that reading the whole thing matters. (If they are, we're going to have a much bigger problem when we try to actually build from it :)

Reply:
Mm yea this is busted, the regex no longer match, I wonder if it's due to the newlines:

https://github.com/google/kythe/pull/2859/commits/d7825ad8d182fb292dc45205afdd5de1cd2d3ba9#diff-f5927cf9c3c71d011a630eb46cdbd0af

Reply:
Ok getting rid of the ^$ checking fixes as expected:

https://github.com/google/kythe/pull/2859/commits/a62f2d4c6000b64b768653d5f401fe6d0d7c2b26